nylas/resources/messages.py

```diff
@@ -0,0 +1,106 @@
+from typing import Dict
+from nylas.handler.api_resources import (
+    ListableApiResource,
+    FindableApiResource,
+    UpdatableApiResource,
+    DestroyableApiResource,
+)
+from nylas.models.messages import (
+    Message,
+    ListMessagesQueryParams,
+    FindMessageQueryParams,
+    UpdateMessageQueryParams,
+    UpdateMessageRequest,
+)
+from nylas.models.response import Response, ListResponse, DeleteResponse
+
+
+class Messages(
+    ListableApiResource,
+    FindableApiResource,
+    UpdatableApiResource,
+    DestroyableApiResource,
+):
+    def list(
+        self, identifier: str, query_params: ListMessagesQueryParams
+    ) -> ListResponse[Message]:
+        """
+        Return all Messages.
+
+        Args:
+            identifier: The identifier of the grant to get messages for.
+            query_params: The query parameters to filter messages by.
+
+        Returns:
+            A list of Messages.
+        """
+        return super(Messages, self).list(
+            path=f"/v3/grants/{identifier}/messages",
+            response_type=Message,
+            query_params=query_params,
+        )
+
+    def find(
+        self, identifier: str, message_id: str, query_params: FindMessageQueryParams
+    ) -> Response[Message]:
+        """
+        Return a Message.
+
+        Args:
+            identifier: The identifier of the grant to get the message for.
+            message_id: The identifier of the message to get.
+            query_params: The query parameters to include in the request.
+
+        Returns:
+            The requested Message.
+        """
+        return super(Messages, self).find(
+            path=f"/v3/grants/{identifier}/messages/{message_id}",
+            response_type=Message,
+            query_params=query_params,
+        )
+
+    def update(
+        self,
+        identifier: str,
+        message_id: str,
+        request_body: UpdateMessageRequest,
+        query_params: UpdateMessageQueryParams,
```

Comment:
Same as above, I don't think there's any query/filter params for updating a message